### PR TITLE
Adding Assignments/Pages/Modules/Module Items or the Content to the iteration

### PR DESCRIPTION
**Pull request by kamijean**

### CODE REVIEW

File: lib/blacklight/models/gradebook.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  class Gradebook
+    def get_pre_data(data, _)
+      categories = get_categories(data)
+      data.search("OUTCOMEDEFINITIONS").children.map do |outcome|
+        content_id = outcome.at("CONTENTID").attributes["value"].value
+        assignment_id = outcome.at("ASIDATAID").attributes["value"].value
+        category_id = outcome.at("CATEGORYID").attributes["value"].value
+        category = categories[category_id]
+        points = outcome.at("POINTSPOSSIBLE").attributes["value"].value
+        { category: category, points: points, content_id: content_id,
```
Comment (bfcoder) [inline lib/blacklight/models/gradebook.rb]: Consider multiline?

File: lib/blacklight/models/wikipage.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module Blacklight
+  class WikiPage < Content
+    def canvas_conversion(course)
+      unless @title == "--TOP--"
+        page = course.pages.
```
Comment (bfcoder) [inline lib/blacklight/models/wikipage.rb]: name this variable page_count?

File: lib/blacklight/xml_parser.rb [hunk 3]
```diff
@@ -11,46 +11,99 @@ module Blacklight
...
+    end
+  end
+
+  def self.get_single_pre_data(pre_data, file)
+    pre_data.detect do |d|
+      d[:file_name] == file || d[:assignment_id] == file
+    end
+  end
+
+  def self.midnight_eagle(resources, zip_file)
```
Comment (bfcoder) [inline lib/blacklight/xml_parser.rb]: lol, nice Jamis.

File: lib/blacklight/xml_parser.rb [hunk 4]
```diff
@@ -11,46 +11,99 @@ module Blacklight
...
+      parent_id = get_master_parent(pre_data, parents_ids,
+                                    content[:parent_id])
+      content[:parent_id] = parent_id
+    end
+  end
+
+  def self.get_master_parent(pre_data, parents_ids, parent_id)
+    parent = pre_data.detect { |p| p[:id] == parent_id }
+    if parents_ids.include? parent[:id]
+      return parent[:id]
```
Comment (bfcoder) [inline lib/blacklight/xml_parser.rb]: nix the return

File: lib/blacklight/xml_parser.rb [hunk 5]
```diff
@@ -11,46 +11,99 @@ module Blacklight
...
+      if gradebook
+        content[:points] = gradebook[:points] || ""
+        content[:assignment_id] = gradebook[:assignment_id] || ""
+      end
+    end
+    pre_data["content"]
+  end
+
+  def self.build_heirarchy(pre_data)
+    parents = pre_data.select { |p| p[:parent_id] == "{unset id}" }
```
Comment (bfcoder) [inline lib/blacklight/xml_parser.rb]: It doesn't appear that parents is used past the next line.
Consider?
```ruby
parents_ids = pre_data.
  select { |p| p[:parent_id] == "{unset id}" }.
  map { |u| u[:id] }
```

**Review by bfcoder — COMMENTED** [review]
Overall code looks good. Just a few comments to consider. Will pull down and test.

File: lib/blacklight/models/gradebook.rb (hunk 1 above)
Comment (kamijean) [reply lib/blacklight/models/gradebook.rb]: What do you mean by this?

Comment (bfcoder) [reply lib/blacklight/models/gradebook.rb]:
```ruby
{
  category: category,
  points: points,
  content_id: content_id,
  assignment_id: assignment_id,
}
```

**Review by bfcoder — APPROVED** [review]
Pulled down and tested. Seems to work well. 👍

**Pull request merged**